Google Auth Library - update to 10.x

There is a type Headers in Google Auth Lib that was removed between [9.15.1](https://github.com/googleapis/google-auth-library-nodejs/blob/v9.15.1/src/auth/oauth2client.ts) to [10.0.0](https://github.com/googleapis/google-auth-library-nodejs/blob/v10.0.0/src/auth/oauth2client.ts)

This breaks the build when the latest version of the lib is used:


 ../node_modules/.pnpm/google-spreadsheet@4.1.5_google-auth-library@10.1.0/node_modules/google-spreadsheet/dist/index.d.ts(5,10): error TS2305: Module '"../../../../../../_thunderstorm/google-services/node_modules/google-auth-library/build/src/auth/oauth2client"' has no exported member 'Headers'.

--- a/.changeset/eleven-fans-bathe.md
+++ b/.changeset/eleven-fans-bathe.md
@@ -1,0 +1,5 @@
+---
+"google-spreadsheet": patch
+---
+
+fix type issue with google-auth-library (fixes #726)

--- a/package.json
+++ b/package.json
@@ -50,7 +50,7 @@
     "test": "vitest",
     "test:ci": "vitest run"
   },
-  "dependencies": {},
+  "dependencies": { },
   "devDependencies": {
     "@changesets/cli": "^2.29.5",
     "@types/node": "^24.2.0",
@@ -64,7 +64,7 @@
     "eslint-config-airbnb-typescript": "^17.0.0",
     "eslint-plugin-import": "^2.27.5",
     "eslint-plugin-no-floating-promise": "^1.0.2",
-    "google-auth-library": "^10.1.0",
+    "google-auth-library": "^10.2.1",
     "ky": "^1.8.2",
     "tsup": "^8.5.0",
     "typescript": "^5.5.4",

--- a/src/lib/GoogleSpreadsheet.ts
+++ b/src/lib/GoogleSpreadsheet.ts
@@ -43,7 +43,14 @@ async function getRequestAuthConfig(auth: GoogleApiAuth): Promise<{
   // JWT | OAuth2Client | GoogleAuth | Impersonate | AuthClient
   if ('getRequestHeaders' in auth) {
     const headers = await auth.getRequestHeaders();
-    return { headers };
+
+    // google-auth-library v10 uses a Headers object rather than a plain object
+    if ('entries' in headers) {
+      return { headers: Object.fromEntries(headers.entries()) };
+    } if (_.isObject(headers)) {
+      return { headers: headers as Record<string, string> };
+    }
+    throw new Error('unexpected headers returned from getRequestHeaders');
   }
 
   // API key only access passes through the api key as a query param
@@ -139,11 +146,7 @@ export class GoogleSpreadsheet {
   async _setAuthRequestHook(req: Request) {
     const authConfig = await getRequestAuthConfig(this.auth);
     if (authConfig.headers) {
-      // google-auth-library v10 uses a Headers object
-      const headers = 'entries' in authConfig.headers
-        ? Object.fromEntries((authConfig.headers as any).entries())
-        : authConfig.headers;
-      Object.entries(headers).forEach(([key, val]) => {
+      Object.entries(authConfig.headers).forEach(([key, val]) => {
         req.headers.set(key, String(val));
       });
     }

--- a/src/lib/types/auth-types.ts
+++ b/src/lib/types/auth-types.ts
@@ -1,11 +1,7 @@
-// optional peer dependency - this ts-ignore should help when user is not using it
-// @ts-ignore
-import type { Headers } from 'google-auth-library/build/src/auth/oauth2client';
-
 /** single type to handle all valid auth types */
 export type GoogleApiAuth =
   // this simple interface should cover all google-auth-library auth methods
-  | { getRequestHeaders: () => Promise<Headers> }
+  | { getRequestHeaders: () => Promise<any> }
   // used to pass in an API key only
   | { apiKey: string }
   // used to pass in a raw token

--- a/tsconfig.json
+++ b/tsconfig.json
@@ -15,8 +15,8 @@
     "resolveJsonModule": true
   },
   "include": [
-    "src",
-    "examples",
+    "src/**/*.ts",
+    "examples/**/*.ts",
     "env.d.ts"
   ]
 }

